exchange/exchange.go

```diff
@@ -220,4 +319,64 @@ func (e *exchange) makeBid(Bids []*pbsOrtbBid) ([]openrtb.Bid, []string) {
 		bids[i].Ext = ext
 	}
 	return bids, errList
+}
+
+// The following may move to /pbs/targeting with pbs/buckets going in there as well. But pbs/buckets in not yet in this branch
+// This also duplicates code in pbs_light, which should be moved to /pbs/targeting. But that is beyond the current
+// scope, and likely moot if the non-openrtb endpoint goes away.
+const (
+	hbpbConstantKey = "hb_pb"
+	hbBidderConstantKey = "hb_bidder"
+	hbSizeConstantKey = "hb_size"
+	hbCreativeLoadMethodConstantKey = "hb_creative_loadtype"
+	hbCreativeLoadMethodHTML = "html"
+	hbCreativeLoadMethodDemandSDK = "demand_sdk"
+	hbCacheIdConstantKey = "hb_cache_id"
+	hbDealIdConstantKey = "hb_deal"
+	)
+
+func min(x, y int) int {
+	if x < y {
+		return x
+	}
+	return y
+}
+
+func (e *exchange) MakePrebidTargets(cpm float64, width uint64, height uint64, cache string, deal string, targData *targetData, adapter openrtb_ext.BidderName) (map[string]string, error) {
```

Comment:
So... I'm a little bit confused by the code organization here, but might just be because I had a different picture in mind. Check out the comment I left here: https://github.com/prebid/prebid-server/blob/master/exchange/bidder.go#L15

The architecture I imagined was:

- `adapters/bidder`: minimum amount of logic needed to reach an external demand source.
- `exchange/bidder`: core logic which only needs to be aware of a _single_ bidder's bids.
- `exchange/exchange`: core logic which needs to be aware of _all_ bidders' bids.

Targeting fits into both of the latter two categories. Some keys are bidder-specific, while others need to know the winning bid.

I'm not sure which is cleaner... so I'll throw out the suggestion and leave it up to you. I think we should either:

- Add params to `pbsOrtbSeatBid`, and possibly introduce a `pbsOrtbBidRequest`, so that the bid-specific targeting can be done in the `exchange/Bidder`. This _might_ improve things because you won't have to `unmarshal/remarshal` so much (since the `ext` hasn't been packaged yet).
- Remove the code comment that I linked above, because it's misleading. Not sure what you'd replace it with, though :/... cause it was the only clear reason I could come up with for dividing the core auction logic like we did.